File: app/src/main/java/com/example/wazkar/recevier/NotificationReceiver.java
package com.example.wazkar.recevier;

import android.app.AlarmManager;
import android.app.Notification;
import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.media.RingtoneManager;
import android.net.Uri;

import androidx.core.app.NotificationCompat;
import androidx.core.app.NotificationManagerCompat;

import com.example.wazkar.MainActivity;
import com.example.wazkar.R;
import com.example.wazkar.data_layer.NotificationDataset;

import java.util.Calendar;

import static android.content.Context.ALARM_SERVICE;

public class NotificationReceiver extends BroadcastReceiver {

    private String CHANNEL_ID="Wazkar";







    @Override
    public void onReceive(Context context, Intent intent1) {


        // TODO notification

        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
        NotificationCompat.Builder builder = new NotificationCompat.Builder(context, CHANNEL_ID);
        builder.setSmallIcon(R.drawable.splash)
                .setContentTitle(context.getString(R.string.notification_title))
                .setAutoCancel(true);


        Intent intent =new Intent(context, MainActivity.class);
        PendingIntent pendingIntent =PendingIntent.getActivity(context,1,intent,PendingIntent.FLAG_UPDATE_CURRENT);
        builder.setContentText(NotificationDataset.getNof());
        builder.setAutoCancel(true);
        builder.setStyle(new NotificationCompat.BigTextStyle());
        Uri uri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
        builder.setSmallIcon(R.drawable.loo);
        builder.setSound(uri);
        builder.setPriority(NotificationCompat.PRIORITY_DEFAULT);








        builder.setContentIntent(pendingIntent);



        Notification notification = builder.build();
        NotificationManagerCompat.from(context).notify(0, notification);





    }
}

File: app/src/main/java/com/islamic/Wazaker/ui/NotificationRecevier/NotificationReceiver.kt
package com.islamic.Wazaker.ui.NotificationRecevier

import android.app.*
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import androidx.core.app.NotificationCompat
import com.islamic.Wazaker.Data.DataSets.NotificationDataset
import com.islamic.Wazaker.ui.Activities.MainActivity.MainActivity
import com.islamic.wazkar.R

class NotificationReceiver : BroadcastReceiver() {

    private val CHANNEL_ID = "Channel 1"

    override fun onReceive(context: Context, intent1: Intent) {

        val notificationIntent = Intent(context, MainActivity::class.java)
        val stackBuilder = TaskStackBuilder.create(context)
        stackBuilder.addParentStack(MainActivity::class.java)
        stackBuilder.addNextIntent(notificationIntent)

        val pendingIntent = stackBuilder.getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT)
        val builder = Notification.Builder(context)

        val notification: Notification = NotificationCompat.Builder(context, CHANNEL_ID)
            .setContentTitle("وذكر")
            .setContentText(NotificationDataset.getRandomNotification())
            .setSmallIcon(R.drawable.splash)
            .setLights(Notification.FLAG_SHOW_LIGHTS, 1000, 500)
            .setDefaults(NotificationCompat.DEFAULT_ALL)
            .setPriority(NotificationCompat.PRIORITY_HIGH)
            .setAutoCancel(true)
            .setContentIntent(pendingIntent)
            .setStyle(NotificationCompat.BigTextStyle())
            .build()

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            builder.setChannelId(CHANNEL_ID)
        }

        val notificationManager =
            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(
                CHANNEL_ID,
                "Channel_ID",
                NotificationManager.IMPORTANCE_HIGH
            )
            channel.enableLights(true)
            channel.enableVibration(false)
            notificationManager.createNotificationChannel(channel)
        }
        notificationManager.notify(0, notification)
    }

}